packages/augur-sdk/src/state/db/DerivedDB.ts

```diff
@@ -195,6 +197,10 @@ export class DerivedDB extends AbstractDB {
     return log;
   }
 
+  protected lock(name: string) {
+    this.locks[name] = true;
+  }
+
   protected async waitOnLock(lock: string, maxTimeMS: number, periodMS: number): Promise<boolean> {
```

Comment:
Just reading this, should we actually let it write if the thing its waiting for exceeds some timeout? I feel like this should actually crash or something before taking an action that would corrupt the DB

Reply:
Great point. Setting it to throw an error instead of returning a bool.